### ISSUE TITLE
WSL 2 setup

### ISSUE THREAD

**Issue by DawidWetzler** [rupor-github/wsl-ssh-agent#18]
Hello there, that's not a real issue but a question. I'm really helpless so I need to ask for your help.
Today I decided to update my windows to version 2004 and use WSL 2 to see if the issues I had with the first version are gone.

What were the steps I have did?
1. Enable OpenSSH in Windows Services
2. Download WSL 2 Workaround files
3. Extract downloaded files (C:\Users\\$USER\\.wsl)
4. Open KeepassXC (v. 2.6.0)
5. Settings → SSH Agent
6. Check following options
	7. Enable SSH Agent integration
	8. Use OpenSSH for Windows instead of pageant
7. Make a simple test in PowerShell (ssh -T git@github.com) (it works)
8. Open WSL and add following to .bashrc
```
export SSH_AUTH_SOCK=$HOME/.ssh/ssh-agent.sock
ss -a | grep -q $SSH_AUTH_SOCK
if [ $? -ne 0   ]; then
            rm -f $SSH_AUTH_SOCK
                ( setsid socat UNIX-LISTEN:$SSH_AUTH_SOCK,fork EXEC:"/mnt/c/Users/$USER/.wsl/npiperelay.exe -ei -s //./pipe/openssh-ssh-agent",nofork & ) >/dev/null 2>&1
fi
```
9. Before restarting wsl I have ran following command in PowerShell:
`.\wsl-ssh-agent-gui.exe -socket C:\Users\$USER\.wsl\ssh-agent.soc`   
10. Check if the paths are correct
```
C:\Users\$USER\.wsl\ssh-agent.sock

\\.pipe\openssh-ssh-agent
```

I thought that the SSH_AUTH_SOCK path may be wrong but even chaning it to the same path as in gui it still does not work. The ssh-agent.sock is being deleted because of the rm -f $SSH_AUTH_SOCK which should be clear.

I really do not know what I am doing wrong, could someone help me and tell me what I am doing wrong?
Thanks in advance.

**Comment by rupor-github** [comment]
Dawid, WSL2 does not actually use `wsl-ssh-agent-gui.exe`, only `npiperelay.exe`, so it does not matter how you started it or how it is configured. 

if you .bashrc is correct and you have installed socat on your WSL2 Linux distro - everything should just work even without you running wsl-ssh-agent-gui.exe from powershell. Socket is being created by socat, when accessed from the WSL2 end socat runs npiperelay.exe which accesses pipe and ssh-agent service on Windows end that it all you need. You could normally validate staff by running `ssh-add -L` from Windows and from WSL2 to see the same results. 

Now - I have no udea what KeypassXC is, what it does and how you want it to be used.

**Comment by DawidWetzler** [comment]
Thanks for such a quick response!

Well, okay, now I understand. Thanks for explaining.

Keepass (KeepassXC) is a simple password manager. While Keepass needs Keeagent (Plugin) to support ssh keys storage KeepassXC doesn't as this functionality is being natively delivered.

While I was on WSL1 I have imported my keys like that: https://gist.github.com/strarsis/e533f4bca5ae158481bbe53185848d49
and it was working fine.  

I've tried running ssh-add -L from Windows and all of my ssh keys have showed up (that's what I expected) while on WSL2 following error occurred:
`error fetching identities: communication with agent failed`.

socat is installed. I don't think that I have messed up with my paths.

I've reinstalled my whole WSL and done all the steps before with leaving the gui thing.
Now ssh-add -L showed all of my keys and it is working. Thank you for your help!
